pandas/tslib.pyx

```diff
@@ -1908,6 +1909,12 @@ cpdef array_to_datetime(ndarray[object] values, raise_=False,
                     iresult[i] = value
                     _check_dts_bounds(&dts)
                 except ValueError:
+                    if require_iso8601:
+                        if coerce:
+                            iresult[i] = iNaT
+                            continue
+                        raise TypeError
```

Comment:
raise a ``ValueError``, but with a meaningul message (which you can then catch above). It is odd to convert to a ``TypeError``